### PR TITLE
[DOR-43][AO] add export mandatory vessel details view

### PR DESCRIPTION
**Pull request by arturopala**

### CODE REVIEW

File: test/uk/gov/hmrc/traderservices/controllers/VesselDetailsFormSpec.scala [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        .updated("timeOfArrival.hour", "00")
+        .updated("timeOfArrival.minutes", "60")
+        .updated("timeOfArrival.period", "ma")
+      form.bind(input).value shouldBe None
+      form.bind(input).errors should haveOnlyErrors(
+        FormError("timeOfArrival", "error.timeOfArrival.invalid-hour-value"),
+        FormError("timeOfArrival", "error.timeOfArrival.invalid-minutes-value"),
+        FormError("timeOfArrival", "error.timeOfArrival.invalid-period-value")
+      )
+    }
```
Comment (Tasmine10ten) [inline test/uk/gov/hmrc/traderservices/controllers/VesselDetailsFormSpec.scala:131]: These tests seem fine though one thing I did notice was just that the error messages for incorrect digits all say "Date" of... etc, should they maybe have their respective names ie. day/month/year. 

(messages file)
image


image

**Review by Tasmine10ten — COMMENTED** [review]
FormFieldMappingSpec
(Line 351)
      mandatoryDateOfArrivalMapping.bind(Map("year" -> "", "day" -> "")) should haveOnlyError[Option[LocalDate]](
        "error.dateOfArrival.required"
      )

I think this will show the 2x: error.dateOfArrival.required-year & -day

**Comment by Tasmine10ten** [comment]
I think we have the content on the spreadsheet so further validation on date can be added for the date field (ie. Date of arrival must be today or in the next 6 months)

**Comment by arturopala** [comment]
> FormFieldMappingSpec
> (Line 351)
> mandatoryDateOfArrivalMapping.bind(Map("year" -> "", "day" -> "")) should haveOnlyError[Option[LocalDate]](%22error.dateOfArrival.required%22)
> 
> I think this will show the 2x: error.dateOfArrival.required-year & -day

in this test case, month value is completely missing from the request, others are empty, so they all are missing, but different ways

**Pull request merged**